### PR TITLE
Update to Go 1.16

### LINKED ISSUE #1387: Update Dockerfile.codegen to remove invalid repo reference
## Summary
Dockerfile.codegen in the repo root tries to clone https://github.com/conjurinc/code-generator.git. This repository does not seem to exist, so the file should be removed or updated.

## Steps to Reproduce
1. Open Dockerfile.codegen and look at the run command.
2. Try to navigate to https://github.com/conjurinc/code-gen

## Expected Results
All files work and don't reference non-existent elements.

## Actual Results (including error logs, if applicable)
404 on accessing repo. Presumably, building an image from the Dockerfile fails as well, though I haven't tried it.

## Additional Information
@izgeri [suggested ](https://conjurhq.slack.com/archives/CQY94LJLB/p1614175562058900?thread_ts=1614127041.052000&cid=CQY94LJLB) that maybe this should be updated to talk to kubernetes/code-generator

### PR DESCRIPTION
**Pull request by szh**
Fixes build failing due to goconvey dependency change. Depends on https://github.com/cyberark/go-mssqldb/pull/30

### Desired Outcome

Jenkins builds of Docker images do NOT fail with errors such as the following:

```
[2021-10-26T21:50:18.036Z] #21 1.547 package embed: unrecognized import path "embed": import path does not begin with hostname
[2021-10-26T21:50:26.134Z] #21 8.793 package io/fs: unrecognized import path "io/fs": import path does not begin with hostname
[2021-10-26T21:50:32.671Z] #21 ERROR: executor failed running [/bin/sh -c go get -u github.com/jstemmer/go-junit-report &&     go get github.com/smartystreets/goconvey]: exit code: 1
```

### Implemented Changes

* The version of Go used in our Docker containers and specified in `go.mod` is upgraded from Go v1.15 to Go v1.16.
* `go env -w GOFLAGS=-mod=mod` is run in containers to fix [this module issue in go1.16](https://github.com/golang/go/issues/44129#issuecomment-865249631)
* Update K8S packages and run `go mod tidy` to fix build breaking due to [this issue](https://github.com/kubernetes/client-go/issues/741)
  * k8s.io/api v0.22.3
  * k8s.io/apiextensions-apiserver v0.22.3
  * k8s.io/apimachinery v0.22.3
  * k8s.io/client-go v0.22.3
* Update arguments to K8S package functions
  * Satisfy new function signatures after K8S package updates.
* Fixes `bin/codegen` script
  * Updates `Dockerfile.codegen` to use correct upsteam `kubernetes/code-generator` repo tag matching K8S packages
  * Fix permission denied deleting temp dirs by deleting within Docker context
* Include autogenerated files after fixing codegen script
* Use go1.16 in GHA macOS runner
* Loosen codeclimate rules for autogenerated code and golang `context.TODO()` false positives
* Use [latest cyberark/go-mssqldb](https://github.com/cyberark/go-mssqldb/tree/3a6ec62f11d6f4443c020481354a15d77a42360d)

### Connected Issue/Story
Incidentally resolves #1387 

### Definition of Done

- Jenkins builds run with no errors.

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

**Comment by szh** [comment]
I'm changing this PR to a draft until cyberark/go-mssqldb#30 is merged in, because I'll have to update the submodule ref.

**Comment by szh** [comment]
Builds failing. Appears to be a Red Hat issue.
From Jenkins:
```
[2021-11-01T15:20:36.580Z] Step 24/37 : FROM registry.access.redhat.com/ubi8/ubi as secretless-broker-redhat

[2021-11-01T15:21:57.956Z] received unexpected HTTP status: 503 Service Unavailable
```

and Github Actions:
```
Step 24/37 : FROM registry.access.redhat.com/ubi8/ubi as secretless-broker-redhat
received unexpected HTTP status: 504 Gateway Time-out
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Changed
+- The project Golang version is updated from the end-of-life v1.15 to version v1.16.
```
Comment (diverdane) [inline CHANGELOG.md:10]: This seems like a change that isn't technically "user facing", and it isn't a security-related change. So I'm thinking that we should leave out this entry in the CHANGELOG.md. Here are our Community guidelines for what should be included in CHANGELOG.md entries:
https://github.com/cyberark/community/blob/main/Conjur/CONTRIBUTING.md#changelog-content

**Review by diverdane — COMMENTED** [review]
Looks good, thanks for taking care of this!
One comment about including an entry in the CHANGELOG.md.

File: CHANGELOG.md (hunk 1 above)
Comment (szh) [reply CHANGELOG.md]: Ok, I removed that.

**Comment by szh** [comment]
Builds seem to have been broken by `go mod tidy` in go-mssqldb. Rolling that back in https://github.com/cyberark/go-mssqldb/pull/31

**Comment by imheresamir** [comment]
Note to merger: Don't squash commits when merging - "Rebase and merge"

**Review by diverdane — APPROVED** [review]
LGTM!!! Thank you so much for taking care of this!

**Pull request merged**